File: BME/hexxagon_java/src/Controls/Move.java
package Controls;

import Enums.MoveType;

import java.io.Serializable;

/**
 * Egy lepest reprezental, ketto koordinatabol all, valamint a lepes tipusabol es ertekebol.
 */
public class Move  implements Serializable {

    public final Coordinate from;
    public final Coordinate to;
    public final MoveType type;
    public int value;

    public Move( Coordinate from,  Coordinate to){

        this.from = from;
        this.to = to;
        type = from.getDistance(to);
    }

    public Boolean isValid(){

        return !type.equals(MoveType.INVALID);
    }

    @Override
    public String toString(){

        return "From: " + from + " | To: " + to;
    }
}

File: own/cpp_projects/chess_console/chess_console/Move.cpp
#include "Move.h"

Move::Move(Coordinate from, Coordinate to)
	:from(from), to(to), jump(false)
{
	if (this->from.x < 0) this->from.x = 0;
	if (this->from.y < 0) this->from.y = 0;
	if (this->to.x < 0) this->to.x = 0;
	if (this->to.y < 0) this->to.y = 0;
}

int Move::distance()
{
	return abs(from.x - to.x) + abs(from.y - to.y);
}
